pointstorm/embedding/text.py

```diff
@@ -33,27 +33,67 @@ class Document(BaseModel):
     text: Optional[list]
     embeddings: Optional[list] = []
 
-def generate_embedding(document: Document, tokenizer: AutoTokenizer, model: AutoModel, model_id: str = None) -> Document:
+def get_openai_models():
+    """
+    Retrieve all currently available embedding models from OpenAI.
+    @params:
+        none
+    returns:
+        An array containing strings of the embedding model names.
+    """
+    try:
+        dotenv.load_dotenv()
+        response = requests.get(
+                url = "https://api.openai.com/v1/models",
+                headers = {
+                        'Authorization': f'Bearer {os.getenv("OPENAI_API_TOKEN")}',
+                        'Content-Type': 'application/json'
+                    }
+            )
+        if response.status_code == 200:
+            data = response.json()['data']
+            return [entry['id'] for entry in data]
+        elif response.status_code == 401:
+            print(f"You didn't provide your API key. You need to provide your API key in an Authorization header using Bearer auth.")
+    except Exception as e:
+        raise e
+
+def generate_embedding(document: Document, tokenizer: AutoTokenizer = None, model: AutoModel = None, embedding_type: str = None, model_id: str = "text-embedding-ada-002") -> Document:
```

Comment:
We should define the embedding type as either huggingface or openai in the example + tests